pkg/kubelet/eviction/helpers.go

```diff
@@ -532,15 +570,24 @@ func makeSignalObservations(summaryProvider stats.SummaryProvider) (signalObserv
 	// build an evaluation context for current eviction signals
 	result := signalObservations{}
 
-	if memory := summary.Node.Memory; memory != nil && memory.AvailableBytes != nil {
-		result[SignalMemoryAvailable] = resource.NewQuantity(int64(*memory.AvailableBytes), resource.BinarySI)
+	if memory := summary.Node.Memory; memory != nil && memory.AvailableBytes != nil && machineInfo != nil {
```

Comment:
i guess in this case, maybe just check if machineInfo != nil && machineInfo.MemoryCapacity != 0 ?

i may be worrying too much.